File: lib/square.rb
class Square
  attr_accessor :left, :right, :up, :down
  attr_reader :x, :y

  def initialize x, y
    @x = x
    @y = y
  end

  def point hsh
    hsh.each do |key, value|
      metaclass.send(:define_method, key) { value }
    end
  end

  def coordinates
    { x: @x, y: @y }
  end

private

  def metaclass
    class << self
      self
    end
  end
end

File: src/square.js
module.exports = Square;

function Square(x, y) {
  var left;
  var right;
  var up;
  var down;

  this.x = x;
  this.y = y;
  this.coordinates = x + ',' + y;

  Object.defineProperties(this, {
    left: {
      get: function() { return left; },
      set: function(square) { left = square; }
    },
    right: {
      get: function() { return right; },
      set: function(square) { right = square; }
    },
    up: {
      get: function() { return up; },
      set: function(square) { up = square; }
    },
    down: {
      get: function() { return down; },
      set: function(square) { down = square; }
    }
  });
}
